Fix test=true checks so that state runs in test mode don't report cha…

prymitive opened the pull request:
…nges if there is nothing to do

Currently running this postgres_extension.present with test=true will always report changes, even if there is nothing to modify.

### What does this PR do?

Fixes `postgres_extension.present` so that it only reports changes in test mode if there are changes to be made.

### What issues does this PR fix or reference?

#44189 

### Tests written?

Yes/No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

prymitive committed:
Fix test=true checks so that state runs in test mode don't report changes if there is nothing to do

Currently running this postgres_extension.present with test=true will always report changes, even if there is nothing to modify.

rallytime commented:
Hi @prymitive and welcome! It looks like this change has caused some tests to fail. Can you take a look? https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu14-n/15974/

Do let us know if you have any questions.

prymitive commented:
Sorry about that, will fix those

prymitive committed:
Fix test case when postgres extension is already present and test=true is passed

Currently test expects that if the extension is already present salt will indicate that it will create it (which won't really happen without test=true since the extension is present), state was fixed so now salt will return correct information with test=true so fix the test to follow correct behaviour.

prymitive commented:
Updated tests to match corrected behaviour, there are new failures but are from `rest_cherrypy` and don't look related to this change.

cachedout commented:
@kiorky Are you around and can you look at this?
